src/executor/contract.rs

```diff
@@ -279,8 +266,25 @@ impl AotContractExecutor {
     pub fn load(library_path: &Path) -> Result<Self> {
         let info_str = std::fs::read_to_string(library_path.with_extension("json"))?;
         let contract_info: NativeContractInfo = serde_json::from_str(&info_str)?;
+
+        let library = Arc::new(unsafe { Library::new(library_path)? });
+        unsafe {
+            let get_version = library
+                .get::<extern "C" fn(*mut u8, usize) -> usize>(b"cairo_native__get_version")?;
+
+            let mut version_buffer = [0u8; 16];
+            let version_len = get_version(version_buffer.as_mut_ptr(), version_buffer.len());
+
+            let target_version = env!("CARGO_PKG_VERSION");
```

Comment:
This would fail if we don't change cargo pkg version, right? Is it possible to use the git rev?

Reply:
Cargo has no notion of git revisions. It'll work between different versions only.

Reply:
Would executing `git rev-parse HEAD` on build time work?

Reply:
Not if we decide to not use git. Since cargo and git are not related (they are independent and know nothing of each other) I don't think it's a good idea to mix them.

Also we'll be depending on the shell environment being properly configured and git installed. You can download the compressed source code without git stuff from the releases page so you don't actually need git for anything; this would break the build for systems that build without git installed.